developer-docs-site/docs/tutorials/full-node/run-a-fullnode.md

```diff
@@ -0,0 +1,310 @@
+* [REST specifications][rest_spec].
+
+:::caution Docker support only on Linux
+
+Docker container is currently supported only on Linux x86-64 platform. If you are on macOS or Windows platform, use the Aptos-core source approach.
+
+:::
+
+## Hardware requirements
+
+We recommend the following hardware resources:
+
+- For running a production grade FullNode:
+
+  - **CPU**: 4 cores (Intel Xeon Skylake or newer).
+  - **Memory**: 8GiB RAM.
+
+- For running the FullNode for development or testing:
+
+  - **CPU**: 2 cores.
+  - **Memory**: 4GiB RAM.
+
+## Storage requirements
+
+The amount of data stored by Aptos depends on the ledger history (length) of the blockchain and the number
+of on-chain states (e.g., accounts). These values depend on several factors, including: the age of the blockchain,
+the average transaction rate and the configuration of the ledger pruner.
+
+:::tip
+
+Given that devnet is currently being reset on a weekly basis, we estimate that Aptos will not require more than several GBs of storage. See the `#devnet-release` channel on Aptos Discord.
+
+:::
+
+## Configuring a FullNode
+
+You can configure a public FullNode in two ways:
+
+1. Using the Aptos-core source code.
+2. Using Docker.
+
+This document describes how to configure your public FullNode using both the methods.
+
+### Using Aptos-core source code
+
+1. Fork and clone the Aptos repo.
+
+    - Fork the Aptos Core repo by clicking on the **Fork** on the top right of this repo page: https://github.com/aptos-labs/aptos-core.
+    - Clone your fork.
+
+      ```
+      git clone https://github.com/<YOUR-GITHUB-USERID>/aptos-core
+
+      ```
+
+2. `cd` into `aptos-core` directory.
+
+    ```
+    cd aptos-core
+    ```
+
+3. Run the `scripts/dev_setup.sh` Bash script as shown below. This will prepare your developer environment.
+
+    ```
+    ./scripts/dev_setup.sh
+    ```
+
+4. Update your current shell environment.
+
+    ```
+    source ~/.cargo/env
+    ```
+
+With your development environment ready, now you can start to setup your FullNode.
+
+5. Checkout the `devnet` branch using `git checkout --track origin/devnet`.
+
+  In the next step you will download and edit the configuration YAML file `public_full_node.yaml` to ensure that your FullNode:
+
+    - Contains the correct genesis blob that is published by the Aptos devnet.
+    - Synchronizes correctly with the devnet, by using the checkpoint file `waypoint.txt` published by the devnet, and
+    - Stores the devnet database at a location of your choice on your local machine.
+
+6. Make sure your current working directory is `aptos-core`. Copy the YAML configuration file from `config/src/config/test_data/public_full_node.yaml` to your current working directory. You will edit this file.
+
+7. The Aptos devnet publishes the `genesis.blob` and `waypoint.txt` files. Download them:
+
+    - Click here [genesis][devnet_genesis] or run the below command on your terminal:
```

Comment:
Missing link here and for waypoint?

Reply:
The `[devnet_genesis]` is defined at the bottom of the page. Link works for both. See the deploy preview: https://deploy-preview-785--aptos-developer-docs.netlify.app/